docs/guides/unit_test_guide.md

```diff
@@ -0,0 +1,70 @@
+ 
+ ### Ginkgo
+  
+ [Ginkgo](https://onsi.github.io/ginkgo/) is one of the most popular framework for writing tests in go.
+ 
+ Read [godoc](https://godoc.org/github.com/onsi/ginkgo) for more information about ginkgo.
+ 
+See a [sample](https://github.com/kubeedge/kubeedge/blob/master/pkg/metamanager/dao/meta_test.go) in kubeedge where go builtin package testing and gomock is used for writing unit tests.
+
+We will soon add a sample in kubeedge where ginkgo is used for writing unit tests.
+
+## Writing UT using GoMock example : metamanager/dao/meta.go
+
+After reading the code of meta.go, we can find that there are 3 interfaces of beego which are used. They are [Ormer](https://github.com/kubeedge/kubeedge/blob/master/vendor/github.com/astaxie/beego/orm/types.go),[QuerySeter](https://github.com/kubeedge/kubeedge/blob/master/vendor/github.com/astaxie/beego/orm/types.go) and [RawSeter](https://github.com/kubeedge/kubeedge/blob/master/vendor/github.com/astaxie/beego/orm/types.go).
+
+We need to create fake implementations of these interfaces so that we do not rely on the original implementation of this interface and their function calls.
+
+Following are the steps for creating fake/mock implementation of Ormer, initializing it and replacing the original with fake.
+1) Create directory mocks/beego
+2) use mockgen to generate fake implementation of the Ormer interface
+```shell
+mockgen -destination=mocks/beego/fake_ormer.go -package=beego github.com/astaxie/beego/orm Ormer
+```
+destination : where you want to create the fake implementation.
+package : package of the created fake implementation file
+github.com/astaxie/beego/orm : the package where interface definition is there
+Ormer : generate mocks for this interface
+
+3) Initialize mocks in your test file. eg meta_test.go
+```shell
+mockCtrl := gomock.NewController(t)
+defer mockCtrl.Finish()
+ormerMock = beego.NewMockOrmer(mockCtrl)
+```
+4) ormermock is now a fake implementation of Ormer interface. We can make any function in ormermock return any value you want.
+5) replace the real Ormer implementation with this fake implementation. DBAccess is variable to type Ormer which we will replace with mock implemention
+```shell
+dbm.DBAccess = ormerMock
+```
+6) If we want Insert function of ormer interface which has return types as (int64,err) to return (1 nil), it can be done in 1 line in your test file using gomock.
+```shell
+ormerMock.EXPECT().Insert(gomock.Any()).Return(int64(1), nil).Times(1)
+```
+Expect() : is to tell that a function of ormermock will be called.
+Insert(gomock.Any()) : expect Insert to be called with any parameter.
+Return(int64(1), nil) : return 1 and error nil
+Times(1): expect insert to be called once and return 1 and nil only once.
+7) So whenever insert is called, it will return 1 and nil , thus removing the dependency on external implementation.
```

Comment:
nit: please check the preview of this line:

```
7) So whenever insert is called, it will return 1 and nil , thus removing the dependency on external implementation.
```

The format is inconsistent with other lines.

LGTM overall except one nit. We can merge this PR once you fix it.